src/preamble.js

```diff
@@ -580,6 +580,9 @@ function removeRunDependency(id) {
 
 Module["preloadedImages"] = {}; // maps url to image data
 Module["preloadedAudios"] = {}; // maps url to audio data
+#if RELOCATABLE
+addOnPreRun(reportUndefinedSymbols);
```

Comment:
why does this run during startup?

Reply:
In case there are undefined symbols in any of the libraries loaded during startup.

We can't run this until all the libraries are loaded since they might mutually depend on each other, but once all them have been loaded there should be no more undefined symbols.